pkgs/data/misc/cacert/default.nix

```diff
@@ -66,6 +66,12 @@ stdenv.mkDerivation {
   setupHook = ./setup-hook.sh;
 
   passthru.updateScript = ./update.sh;
+  passthru.tests = {
+    # Test that building this derivation with a blacklist works, and that UTF-8 is supported.
+    blacklist-utf8 = (cacert.override {
+      blacklist = [ "CFCA EV ROOT" "NetLock Arany (Class Gold) Főtanúsítvány" ];
```

Comment:
Picking up the discussion started on IRC: How can we ensure that the certificates that are blacklisted are actually being execuded or in other worse that they are part of the input set.

My rough idea would be to do it like this:

```nix
{
blacklist-utf8 =
let
  blacklist = [ "CFCA EV ROOT" "NetLock Arany (Class Gold) Főtanúsítvány" ];
in runCommand "verify-the-cacert-filter-output" {
  cacertWithExludes = cacert.override { inherit blacklist; };
  inherit cacert;
} ''
  ${lib.concatStrigsSepMap "\n" (entry: "grep -q '${entry}'  $cacert || ({ echo "CA ${entry} is not part of the ca certificates (anymore). Consider picking another candidate." >&2; exit 1; })")}

  ${lib.concatStrigsSepMap "\n" (entry: "grep -v -q '${entry}'  $cacertWithExcludes || ({ echo "CA ${entry} is *still* part of the ca certificates. Our filtering code seems to be broken." >&2; exit 1; })")}
'';
}
```